plugin/qrfilter.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file qrfilter.cpp
+ * @copyright 2020 Open Mobile PLatform Ltd.
+ * @author Dmitry Butakov d.butakov@omprussia.ru
+ */
+
+#include "qrfilter.h"
+#include "videofilterrunnable.h"
+
+QrFilter::QrFilter(QObject *parent)
+    : QAbstractVideoFilter(parent)
+{
+    connect(this, &QrFilter::activeChanged, this, [=]{
```

Comment:
By default filters are active. This one has to be off by default.

BTW, how `active` property is being manipulated? Couldn't figure it out from the first glance.

Reply:
I will try to argue the need to leave the filter active by default. Imagine that the user opens the camera on Sailfish for the first time and is like this: "Hmm, I wonder if the camera application reads QR", and with a greater degree of probability he will not climb to look into the settings (since this will also be new actions for him), but will his usual action is to find QR at hand. And after he was convinced of the presence of a feature, only then he will have a question: "Okay, and if I don't need this, how can I turn it off" and then he has a more significant goal to go into the settings. From the point of view of an ordinary user, this is a simpler behavior.

`active` property is being manipulated from camera ui, depending on source (front camera/main camera), capture mode (photo/video), setting value (enabled/disabled).

Reply:
The chance of somebody accidentally focusing on a QR code, discovering this feature and finding it useful, is negligible. The price for that would be extra CPU/resource usage, shuttering, lost frames, waste of battery life, and overall degradation of user experience for everyone else. I don't think it's a fair trade-off.

Attention to the new feature can be attracted with a hint, it the feature is buried too deeply into the settings. If it's part of the easily accessible overlay camera menu, I wouldn't even do that.

Reply:
Not sure do I interpret this correctly, but I read above as whether QR code detection as an application feature should be off by default. That would be an option to consider, but it shouldn't affect the code here at all. The filter will not apply unless the application installs one, and the app has control whether it's enabled or not.

Reply:
Oh, it's better then. I was under impression that this filter would get automatically loaded into every process that starts camera.

But still, even if you have to load it explicitly, and it automatically spawns a D-Bus service and starts capturing/analyzing the frames until you turn it off, that would slow down the startup and waste certain amount of system resources on every start, regardless of whether this feature is on or off. It's far more efficient to have it disabled at startup and then turn it on when/if it's needed.

You can still have it on by default at the application level, if you like the idea of some people accidentally focusing on a QR code, discovering the new feature and finding it useful (and other people wondering why running the Camera app now makes the device warmer than usual)

Reply:
Oh, it's already merged 🤔
Never mind then.